File: qb-engineer-ui/src/app/features/shipments/models/create-shipment-request.model.ts
import { CreateShipmentLineRequest } from './create-shipment-line-request.model';

export interface CreateShipmentRequest {
  salesOrderId: number;
  shippingAddressId?: number;
  carrier?: string;
  trackingNumber?: string;
  shippingCost?: number;
  weight?: number;
  notes?: string;
  lines: CreateShipmentLineRequest[];
}

File: qb-engineer-server/qb-engineer.core/Models/CreateShipmentRequestModel.cs
namespace QBEngineer.Core.Models;

public record CreateShipmentRequestModel(
    int SalesOrderId,
    int? ShippingAddressId,
    string? Carrier,
    string? TrackingNumber,
    decimal? ShippingCost,
    decimal? Weight,
    string? Notes,
    List<CreateShipmentLineModel> Lines);

public record CreateShipmentLineModel(
    int? SalesOrderLineId,
    int Quantity,
    string? Notes,
    int? PartId = null);
